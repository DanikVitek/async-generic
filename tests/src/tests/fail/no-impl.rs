use async_generic::async_generic;

trait Trait {
    // ERROR: async_generic can not be applied to traits
    fn do_stuff() {}
}

struct Struct {}

#[async_generic]
impl Trait for Struct {}

fn main() {}
