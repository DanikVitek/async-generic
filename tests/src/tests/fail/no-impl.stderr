error: async_generic can only be used with functions
  --> src/tests/fail/no-impl.rs:10:1
   |
10 | #[async_generic]
   | ^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `async_generic` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected `fn`
  --> src/tests/fail/no-impl.rs:11:1
   |
11 | impl Trait for Struct {}
   | ^^^^
